runtime/services/autocli.go

```diff
@@ -61,6 +62,36 @@ func ExtractAutoCLIOptions(appModules map[string]interface{}) map[string]*autocl
 				moduleOptions[modName] = modOptions
 			}
 		}
+
+		if mod, ok := mod.(appmodule.HasServices); ok {
+			// try to auto-discover options based on the last msg and query
+			// services registered for the module
+			cfg := &autocliConfigurator{}
+			err := mod.RegisterServices(cfg)
+			if err != nil {
+				panic(err)
+			}
+			modOptions := &autocliv1.ModuleOptions{}
+			haveServices := false
+
+			if cfg.msgServer.serviceName != "" {
+				haveServices = true
+				modOptions.Tx = &autocliv1.ServiceCommandDescriptor{
+					Service: cfg.msgServer.serviceName,
+				}
+			}
+
+			if cfg.queryServer.serviceName != "" {
+				haveServices = true
+				modOptions.Query = &autocliv1.ServiceCommandDescriptor{
+					Service: cfg.queryServer.serviceName,
+				}
+			}
+
+			if haveServices {
+				moduleOptions[modName] = modOptions
+			}
+		}
```

Comment:
I think we can reduce code duplication a lot here

Reply:
Done, I've also did the correct implementation of RegisterServices in the autocli configurator. We should be good to merge now 👌

Reply:
Done, I've also did the correct implementation of RegisterServices in the autocli configurator. We should be good to merge now 👌